rllib/algorithms/algorithm.py

```diff
@@ -1774,17 +1890,74 @@ def import_policy_model_from_h5(
-            checkpoint_dir, "checkpoint-{}".format(self.iteration)
-        )
-        pickle.dump(self.__getstate__(), open(checkpoint_path, "wb"))
+        """Exports AIR Checkpoint to a local directory and returns its directory path.
+
+        The structure of an Algorithm checkpoint dir will be as follows::
+
+            policies/
+                pol_1/
+                    policy_state.pkl
+                pol_2/
+                    policy_state.pkl
+            checkpoint_version.txt
+            state.pkl
 
-        return checkpoint_path
+        Note: `checkpoint_version.txt` contains a version string (e.g. "v0") helping
+        RLlib to remain backward compatible wrt restoring from checkpoints from
+        Ray 2.0 onwards.
+
+        Args:
+            checkpoint_dir: The directory where the checkpoint files will be stored.
+
+        Returns:
+            The path to the created AIR Checkpoint directory.
+        """
+        state = self.__getstate__()
+
+        # Extract policy states from worker state (Policies get their own
+        # checkpoint sub-dirs).
+        # TODO: "state" key inside "worker" should be renamed to "policy_states".
+        policies = {}
+        if hasattr(self, "workers") and isinstance(self.workers, WorkerSet):
+            policies = state["worker"].pop("state", {})
+
+        # Add RLlib checkpoint version.
+        state["__version__"] = "v1"
+
+        # Write state (w/o policies) to disk.
+        state_file = os.path.join(checkpoint_dir, "state.pkl")
+        pickle.dump(state, open(state_file, "wb"))
+        # Write checkpoint version separately. This file will NOT be used
+        # by RLlib anywhere, it is solely for the user's convenience.
+        with open(os.path.join(checkpoint_dir, "checkpoint_version.txt"), "w") as f:
+            f.write(state["__version__"])
+
+        # Write individual policies to disk.
+        if hasattr(self, "workers") and isinstance(self.workers, WorkerSet):
+            for pid, policy in self.workers.local_worker().policy_map.items():
+                policy_dir = os.path.join(checkpoint_dir, "policies", pid)
+                os.makedirs(policy_dir, exist_ok=True)
+                # Write policy state.
+                policy_state_file = os.path.join(policy_dir, "policy_state.pkl")
+                pickle.dump(policies[pid], open(policy_state_file, "wb"))
+                # Write model files, if required.
+                if self.config["checkpoints_contain_native_model_files"]:
+                    model_dir = os.path.join(policy_dir, "model")
+                    os.makedirs(model_dir, exist_ok=True)
+                    policy.export_model(export_dir=model_dir)
+
+        return checkpoint_dir
 
     @override(Trainable)
-    def load_checkpoint(self, checkpoint_path: str) -> None:
-        extra_data = pickle.load(open(checkpoint_path, "rb"))
-        self.__setstate__(extra_data)
+    def load_checkpoint(self, checkpoint: Union[Dict, str]) -> None:
```

Comment:
I don't get why this API needs to accept a state dict?
I can imagine it accepts a str path (to a checkpoint dir), or a Checkpoint instance.
don't quite get the utility of a Dict input here.

Reply:
ask: tune.Trainable

Not sure. I think you are supposed to do:
```
algo = PPO(config=config)
algo.restore([some AIR Checkpoint obj].to_dictionary())
```
?

Reply:
-> `tune.Trainable.restore()` calls `Algorithm.load_checkpoint()` under the hood.